app/serializers/current_user_serializer.rb

```diff
@@ -312,7 +313,15 @@ def has_topic_draft
     true
   end
 
+  def draft_count
+    Draft.stream({ user: object, offset: 0, limit: 30 }).length
```

Comment:
Using `.length` in ActiveRecord will cause all of the objects to be loaded into memory. What you want here is `.count` which will do an SQL count query.

Reply:
Also just took a look at draft.stream and in this case you don't care about limit and offset, that is just used for pagination. Looking at the stream method:

```
def self.stream(opts = nil)
  opts ||= {}

  user_id = opts[:user].id
  offset = (opts[:offset] || 0).to_i
  limit = (opts[:limit] || 30).to_i

  stream = Draft.where(user_id: user_id)
    .order(updated_at: :desc)
    .offset(offset)
    .limit(limit)

  # Preload posts and topics to avoid N+1 queries
  Draft.preload_data(stream, opts[:user])

  stream
end
```

I think what you should do is add a new option here called `:all` or `:no_limit`, and then you could use that and don't do the limit and offset. So the code becomes like this:

```ruby
stream = Draft.where(user_id: user_id).order(updated_at: :desc)

if !opts[:all]
   stream = stream.offset(offset).limit(limit)
end
```

And your call becomes:

```
Draft.stream({ user: object, all: true }).count
```